scripts/ghidra/PatchestryDecompileFunctions.java

```diff
@@ -156,6 +203,167 @@ private HighVariable lValueOf(Varnode node) throws Exception {
+            for (int i = 0; i < arguments.length; i++) {
+                beginObject();
+                name("name").value(arguments[i].getName());
+                name("size").value(arguments[i].getLength());
+                name("type").value(label(arguments[i].getDataType()));
+                endObject();
+            }
+            endArray();
+        }
+
+        private void serialize(DataType data_type) throws Exception {
+            if (data_type == null) {
+                nullValue();
+                return;
+            }
+
+            if (data_type instanceof Pointer) {
+                serializePointerType((Pointer) data_type);
+            } else if (data_type instanceof TypeDef) {
+                serializeTypedefType((TypeDef) data_type);
+            } else if (data_type instanceof Array) {
+                serializeArrayType((Array) data_type);
+            } else if (data_type instanceof Structure) {
+                serializeCompositeType((Composite) data_type, "struct");
+            } else if (data_type instanceof Union) {
+                serializeCompositeType((Composite) data_type, "union");
+            } else if (data_type instanceof AbstractIntegerDataType){
+                serializeBuiltinType(data_type, "integer");
+            } else if (data_type instanceof AbstractFloatDataType){
+                serializeBuiltinType(data_type, "float");
+            } else if (data_type instanceof BooleanDataType){
+                serializeBuiltinType(data_type, "boolean");
+            } else if (data_type instanceof Enum) {
+                serializeBuiltinType(data_type, "enum");
+            } else if (data_type instanceof VoidDataType) {
+                serializeBuiltinType(data_type, "void");
+            } else if (data_type instanceof Undefined || data_type.toString().contains("undefined")) {
+                serializeBuiltinType(data_type, "undefined");
+            } else if (data_type instanceof FunctionDefinition) {
+                serializeFunctionDefinition((FunctionDefinition) data_type);
+            } else {
+                throw new Exception("Unhandled type: " + data_type.toString());
+            }
+        }
+
+        private void serializeTypes() throws Exception {
+            for (int i = 0; i < types_to_serialize.size(); i++) {
+                DataType type = types_to_serialize.get(i);
+                name(label(type)).beginObject();
+                serialize(type);
+                endObject();
+            }
+
+            println("Total serialized types: " + types_to_serialize.size());
+        }
+
+        private void serialize(FunctionPrototype proto) throws Exception {
+            if (proto == null) {
+                nullValue();
+                return;
+            }
+
+            name("parameters").beginArray();
+            for (int i = 0; i < proto.getNumParams(); i++) {
+                serialize(proto.getParam(i).getHighVariable());
```

Comment:
Is it possible to have a parameter type and not a high variable? The logic of `serialize` on a `HighVariable` below suggests that if we fail to get the high variable then we also will fail to get the type, but does that have to be true? I think you should merge the `serialize(HighVariable)` logic right into this loop, as the below high variable serialization logic is not general enough to handle cases like whether or not its a global, a local, etc.

Reply:
Inlined the serialization method for paramaters and added an assert check if high variable is not local.